core/src/Kokkos_CopyViews.hpp

```diff
@@ -3478,7 +3478,9 @@ void check_view_ctor_args_create_mirror() {
 // create a mirror
 // private interface that accepts arbitrary view constructor args passed by a
 // view_alloc
-template <class T, class... P, class... ViewCtorArgs>
+template <class T, class... P, class... ViewCtorArgs,
+          class = std::enable_if_t<
+              std::is_void<typename ViewTraits<T, P...>::specialize>::value>>
```

Comment:
`specialize` only matters for the public interface. Some `Trilinos` packages `specialize` `create_mirror` for their purposes and so it's important that `create_mirror_view` calls `create_mirror` (and not any function in the `Impl` namespace).

Reply:
Ok, I will reintroduce the `specialize` check for the public functions only.

I didn't get from the [documentation](https://kokkos.org/kokkos-core-wiki/API/core/view/view.html?highlight=specialize#_CPPv410specialize) what was the purpose of this tag.

Reply:
Ideally, yes. It's up to you if you want to fix it here or not.

Reply:
I think it's ok to generalize it here.